tests/framework/common.go

```diff
@@ -133,6 +144,8 @@ func registerFlags(td *OsmTestData) {
 	flag.StringVar((*string)(&td.CollectLogs), "collectLogs", string(CollectLogsIfErrorOnly), "Defines if/when to collect logs.")
 
 	flag.StringVar(&td.ClusterName, "kindClusterName", "osm-e2e", "Name of the Kind cluster to be created")
+	flag.StringVar(&td.ClusterOS, "ClusterOS", DefaultClusterOS, "The operating system of the working nodes in the cluster. Currently only linux is supported.")
```

Comment:
I'm not sure if having this field is necessary. To me this implies that the tests are configurable based on the value supplied here which isn't the case right now. At the moment we only have Linux e2es and we can't run Linux images on Windows machines. Eventually we'd need to add some abstraction where we can deploy clients/servers on Windows or Linux based on some flag the user passes in, but that seems outside the scope of just adding nodeSelectors.

This flag could be used in the future when we have Windows e2es and hybrid e2es and we want the ability to specify which ones to run, but we're not there yet.

It's still necessary to include the nodeSelectors because these tests should be able to run on hybrid clusters. In that case, I think it'll be best to make this change as minimal as possible right now by keeping the OS parameter for the pod/deployment definitions. But it should be explicitly programmed to correspond to the container image supplied, and not configurable based on how the test is run.

Reply:
removed the flag with the latest commit